tests/Feature/FlightControllerTest.php

```diff
@@ -0,0 +1,403 @@
+use App\Models\City;
+use App\Models\Flight;
+use Database\Seeders\AirlineSeeder;
+use Database\Seeders\CitySeeder;
+use Database\Seeders\DatabaseSeeder;
+use Database\Seeders\FlightSeeder;
+use DateTime;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use PHPUnit\Framework\Attributes\DataProvider;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Date;
+use Tests\TestCase;
+
+class FlightControllerTest extends TestCase
+{
+    use RefreshDatabase;
+    private $baseUrl = '/api/flights';
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->seed([DatabaseSeeder::class]);
+    }
+
+    /** @test */
+    public function it_should_return_a_list_of_flights(): void
+    {
+        $response = $this->get($this->baseUrl);
+
+        $response->assertStatus(200);
+        $response->assertJsonStructure([
+            'data' => [
+                '*' => [
+                    'id',
+                    'airline',
+                    'origin_city',
+                    'destination_city',
+                    'departure_at',
+                    'arrival_at'
+                ],
+            ],
+        ]);
+    }
+
+    /** @test */
+    public function it_should_return_a_list_of_flights_paginated_and_include_pagination_data(): void
+    {
+        $response = $this->get("{$this->baseUrl}?page=2");
+
+        $response->assertStatus(200);
+        $response->assertJsonStructure([
+            'data' => [
+                '*' => [
+                    'id',
+                    'airline',
+                    'origin_city',
+                    'destination_city',
+                    'departure_at',
+                    'arrival_at'
+                ],
+            ],
+            'current_page',
+            'from',
+            'last_page',
+            'path',
+            'per_page',
+            'to',
+            'total',
+        ]);
+        $response->assertJson([
+            'current_page' => 2,
+            'per_page' => 10,
+            'total' => 20,
+        ]);
+    }
+
+    /** @test */
+    public function it_should_return_an_empty_list_of_flights_when_database_is_empty(): void
+    {
+        Flight::truncate();
+
+        $response = $this->get($this->baseUrl);
+
+        $response->assertStatus(200);
+        $response->assertJson(['data' => []]);
+        $response->assertJson(['total' => 0]);
+    }
+
+    /** @test */
+    public function it_should_store_a_new_flight(): void
+    {
+        $flightData = Flight::factory()->make([
+            'departure_at' => '2023-09-10',
+            'arrival_at' => '2023-09-11',
+        ])->toArray();
+
+        $response = $this->postJson($this->baseUrl, $flightData);
+
+        $response->assertStatus(JsonResponse::HTTP_CREATED);
+        $response->assertJson([
+            'message' => 'Flight stored.',
+            'status' => 'success',
+        ]);
+        $this->assertDatabaseHas('flights', $flightData);
```

Comment:
```suggestion
        $this->assertDatabaseHas(Flight::class, $flightData);
```
You can pass a model class to this method and it will infer the table name.